chia/wallet/wallet.py

```diff
@@ -623,10 +629,70 @@ async def get_coins_to_offer(
         return await self.select_coins(amount, min_coin_amount=min_coin_amount, max_coin_amount=max_coin_amount)
 
     # WSChiaConnection is only imported for type checking
-    async def coin_added(
-        self, coin: Coin, height: uint32, peer: WSChiaConnection
-    ) -> None:  # pylint: disable=used-before-assignment
-        pass
+    async def coin_added(self, coin: Coin, height: uint32, peer: WSChiaConnection) -> None:
+        wallet_node = self.wallet_state_manager.wallet_node
+        coin_states: Optional[List[CoinState]] = await wallet_node.get_coin_state([coin.parent_coin_info], peer=peer)
```

Comment:
This duplicated fetching of spend info and metadata matching/parsing can be avoided with some refactoring.